### ISSUE TITLE
Android showing up as a laptop in devices view

### ISSUE THREAD

**Issue by rfk** [mozilla/fxa#240]
Reported by a user in IRC:

```
i deleted my firefox account, made a new one with a new password, but same email address. synced first device: linux firefox 52, and synced everything, restarted this linux firefox, waited for anything else to sync, then downloaded installed firefox mobile 55 on android, and signed in, and while im waiting for it to sync, i look in devices on my sync account and i see it listed as "Firefox on
SM-G935T"
as a laptop
with a laptop icon
and i can see "Firefox for Android" under it, saying "none connected"
whats the deal here
when i go to sync management on firefox mobile, its also showing my current device as a laptop icon as "Firefox on SM-G935T"
with the same thing about "none connected" for "Firefox for Android"
what gives
```

Possibly a regression in the Android client code, or perhaps in our ability to guess the correctly device type?

**Comment by rfk** [comment]
A bit of additional info:

```
it was on both android nightly 55 and android firefox 52, both on new firefox sync accounts, with initial device added onto sync account being a linux firefox 52, then nightly/stable mobile added, one at a time to a whole new firefox sync account (hope that makes sense of what i did)
```

**Comment by azizLIGHT** [comment]
Just to clarify, I made a new Firefox sync account on a Linux Firefox 52, signed in, selected to sync everything except passwords, waited for everything to sync to the account, restarted Linux Firefox 52, waited for anything else to sync, then I added Android Nightly 55 to the sync account, and then noticed on the device list (on both Linux and Android devices), that my Android device is listed as "Nightly on SM-G935T" but with a laptop icon, and that "none connected" for "Firefox on Android" in the list where there's actually a phone icon.

Here's a screenshot of the device list at that point:
![bflpb8d](https://cloud.githubusercontent.com/assets/8013867/24850024/68a0d77c-1d9c-11e7-94ec-f5a353ad38cf.png)


I also repeated this process again after deleting the sync account and making a new one with the same email address but different password to the sync account, and added the same Linux Firefox 52 and repeated the same process but instead of adding Android Nightly 55, I tested it with Android Firefox 52 instead. Then, the device was listed as "Firefox on SM-G935T" with the same laptop icon and also showing "none connected" for "Firefox on Android" in the device list.

I've repeated this process 4 times already.
